builtin/config.c

```diff
@@ -213,8 +243,10 @@ struct strbuf_list {
 
 static int format_config(struct strbuf *buf, const char *key_, const char *value_)
 {
-	if (show_origin)
+	if (show_origin) {
 		show_config_origin(buf);
+		show_config_origin_type(buf);
```

Comment:
I am not quite sure that we can change the output of `git config`, in particular with `-z` (which is intended for scripts or other applications to consume, and which would be broken by this change)...